hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientExecutorServiceProxy.java

```diff
@@ -573,7 +643,9 @@ public void onFailure(Throwable t) {
         private final Map<Member, Object> values;
         private final AtomicInteger members;
 
-        private MultiExecutionCallbackWrapper(int memberSize, MultiExecutionCallback multiExecutionCallback) {
+        private MultiExecutionCallbackWrapper(int memberSize,
+                                              @Nonnull MultiExecutionCallback multiExecutionCallback) {
+            checkNotNull(multiExecutionCallback, "multiExecutionCallback must not be null");
```

Comment:
Aligning behaviour with member-side. NPE should be thrown as soon as the input is not valid and not later in a separate thread.